core/src/main/java/org/apache/iceberg/stats/StatsReader.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.stats;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.AbstractMap.SimpleEntry;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.iceberg.io.InputFile;
+import org.apache.iceberg.io.SeekableInputStream;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.io.ByteStreams;
+
+public class StatsReader implements Closeable {
+
+  private final long fileLength;
+  private final SeekableInputStream input;
+  private Optional<Integer> knownFooterSize;
+  private Optional<FileMetadata> knownFileMetadata = Optional.empty();
+
+  public StatsReader(InputFile inputFile, Optional<Integer> footerSize) {
```

Comment:
What is expected for `footerSize`? Should that match `FooterPayloadSize` or be larger by some constant to account for magic and other non-payload bytes?

Reply:
size of Footer (in bytes). Footer is defined in the file format spec https://github.com/apache/iceberg-docs/pull/69 
````
### Footer structure

Footer has the following structure

```
Magic FooterPayload FooterPayloadSize Reserved Flags FileFormatVersion Magic
```
````

Reply:
After going through the code, it's clear that this is the footer payload size + a constant. This should be documented in Javadoc.

Reply:
yes, this is  footer payload size + a constant, but a user doesn't control "footer payload size" on its own

Footer size can be obtained 

- from manually inspecting the file structure, and `Footer` 's size this is
- from stats writer, as it exposes this information for convenience

I can surely add some javadoc but i don't know what it should contain besides "@param footerSize the size of the file footer defined in the file format specification". Please advise.